pandas/_libs/src/klib/khash_python.h

```diff
@@ -330,6 +332,52 @@ Py_hash_t PANDAS_INLINE tupleobject_hash(PyTupleObject* key) {
 }
 
 
+// TODO: same thing for timedelta64 objects
+Py_hash_t np_datetime64_object_hash(PyDatetimeScalarObject* key) {
```

Comment:
I think you can move this to the capsule. It would fit much more naturally there than the khash header

Reply:
The capsule also already does the PyDateTime_IMPORT, which I think will help your remaining issue